subsys/bluetooth/host/id.c

```diff
@@ -93,6 +93,18 @@ static void adv_is_private_enabled(struct bt_le_ext_adv *adv, void *data)
 	}
 }
 
+static void adv_is_rpa_addr_generated_for_adv_id(struct bt_le_ext_adv *adv, void *data)
+{
+	struct bt_adv_id_check_data *check_data = data;
+
+	if (check_data->id == adv->id &&
+		atomic_test_bit(adv->flags, BT_ADV_RPA_VALID)) {
```

Comment:
Why are you using `BT_ADV_RPA_VALID` instead of `BT_ADV_ENABLED` to set the `adv_enabled` bool?